parachain/pallets/erc20-app/src/lib.rs

```diff
@@ -88,19 +86,15 @@ decl_module! {
 
 			T::Assets::withdraw(AssetId::Token(token_addr), &who, amount)?;
 
-			let message = OutPayload {
+			let message = OutboundPayload {
 				token_addr: token_addr,
 				sender_addr: who.clone(),
 				recipient_addr: recipient,
 				amount: amount
 			};
-			T::Commitments::add(Self::address(), message.encode());
-
+			T::MessageCommitment::add(Self::address(), 0, &message.encode());
```

Comment:
How do we deal with nonces here? This `0` is static, but should be incrementing, right?

Reply:
That was just a placeholder for the nonce handling I've implemented in this follow-up PR: https://github.com/Snowfork/polkadot-ethereum/pull/223/files#diff-a3f0c878bcb36ffd0b0fabe45b220da7797857e800a53208b5fce46161dd50a4R41

The outbound channel in the above PR is responsible for incrementing the nonce.

Reply:
Cool, thanks for clarifying 👍